File: modules/wem-api/src/main/java/com/enonic/wem/api/form/Layout.java
package com.enonic.wem.api.form;


public abstract class Layout
    extends FormItem
{
    Layout( final String name )
    {
        super( name );
    }


    @Override
    FormItemPath resolvePath()
    {
        return resolveParentPath();
    }

    public abstract FormItem getFormItem( final String name );
}

File: modules/wem-webapp/src/main/webapp/admin/common/js/form/Layout.ts
module api.form {

    export class Layout extends FormItem {

        constructor(name: string) {
            super(name);
        }

        public toLayoutJson(): api.form.json.FormItemTypeWrapperJson {

            if (this instanceof FieldSet) {
                return (<FieldSet>this).toFieldSetJson();
            }
            else {
                throw new Error("Unsupported Layout: " + this);
            }
        }

        equals(o: api.Equitable): boolean {

            if (!(o instanceof Layout)) {
                return false;
            }

            return super.equals(o);
        }
    }
}